internal/user/repository.go

```diff
@@ -97,3 +100,53 @@ func (r *Repository) GetUserTypeByEmail(email string) (string, error) {
 	}
 	return userType, nil
 }
+
+// CreateUserWithEmail creates a new user with given email and user_type
+// returns the user struct, and an error
+func (r *Repository) CreateUserWithEmail(email, user_type string) (User, error) {
+	u := User{}
+
+	userID, err := ksuid.NewRandom()
+	if err != nil {
+		return u, err
+	}
+
+	u.ID = userID.String()
+	u.Email = email
+	u.CreatedAt = time.Now()
+	u.Type = user_type
+	u.CreatedAtHumanised = humanize.Time(u.CreatedAt.UTC())
+	if _, err := r.db.Exec(`INSERT INTO users (id, email, created_at, user_type) VALUES ($1, $2, $3, $4)`, u.ID, u.Email, u.CreatedAt, u.Type); err != nil {
+		return User{}, err
+	}
+
+	return u, nil
+}
+
+// GetUserTypeByEmailOrCreateUserIfRecruiter seeks a user in the users table or creates one and recruiter profile if the user has a job posting
+// returns user struct and an error
+func (r *Repository) GetUserTypeByEmailOrCreateUserIfRecruiter(email string, jobRepo *job.Repository, recRepo *recruiter.Repository, svr server.Server) (string, error) {
```

Comment:
```suggestion
func (r *Repository) GetUserTypeByEmailOrCreateUserIfRecruiter(email string, jobRepo *job.Repository, recRepo *recruiter.Repository) (string, error) {
```

remove server from repository please